Civi/Api4/Query/Api4SelectQuery.php

```diff
@@ -468,11 +484,17 @@ protected function composeClause(array $clause, string $type) {
       }
     }
 
-    $sql_clause = \CRM_Core_DAO::createSQLFilter($fieldAlias, [$operator => $value]);
-    if ($sql_clause === NULL) {
-      throw new \API_Exception("Invalid value in $type clause for '$expr'");
+    if ($operator === 'IS EMPTY' || $operator === 'IS NOT EMPTY') {
+      // If field is not a string or number, this will pass through and use IS NULL/IS NOT NULL
+      $operator = str_replace('EMPTY', 'NULL', $operator);
+      if (in_array($field['data_type'] ?? NULL, ['String', 'Integer', 'Float'], TRUE)) {
+        $emptyVal = $field['data_type'] === 'String' ? '""' : '0';
+        $isEmptyClause = $operator === 'IS NULL' ? "= $emptyVal OR" : "<> $emptyVal AND";
+        return "$fieldAlias $isEmptyClause $fieldAlias $operator";
```

Comment:
I think you need to put () around this - although it didn't turn out to be the trick to making my search work

https://lab.civicrm.org/documentation/docs/user-en/-/merge_requests/467

Reply:
@colemanw if you add the () I'll merge this  - if we get more feedback we can tinker further with the labels but I think the bulk of the code is not going to change. I put a screenshot with empty into the docs

Reply:
@eileenmcnaughton did you see any problematic sql caused by the lack of brackets? It appears to me like they're already getting added by the api4 query builder:
![image](https://user-images.githubusercontent.com/2874912/110386634-88fc0200-802e-11eb-9125-4a70a7aa5409.png)

Reply:
@colemanw it was in the JOIN clauses I was not seeing brackets added - ie where you join on source IS EMPTY + some other critieria there are no parentheses

Reply:
Well that's inconsistent. Ok I've added the brackets @eileenmcnaughton and maybe someday we'll clean that up.